### PR TITLE
Add Student Action Logs Table and User Access Control

### PR DESCRIPTION
**Pull request by owl12358**
Screen Shot 2022-05-03 at 4 15 05 PM
Screen Shot 2022-05-03 at 4 15 57 PM
Screen Shot 2022-05-03 at 4 15 48 PM
Screen Shot 2022-05-03 at 4 15 12 PM

Able to communicate with the StackMaps API. 
Using Faraday for external communication
Using Devise gem for user accounts
Need tests, and documentation
**Note: DB must be seeded with initial superuser in order to get general access to this app in prod!** @orsted51 
Also note: current user functionality is temporary and very insecure.
cheetolock

### CODE REVIEW

File: app/controllers/concerns/http_connection.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  ACTION_TYPES = 'studentactionlogs_action_types_ro/'
+  MODULE_NAMES = 'cohorts_ro/'
+  STUDENT_LOGS = 'studentactionlogs_student_data_ro/'
+
+  def stackmaps_api
+    Faraday.new(
+      url: 'https://www.stackmaps.com/api/',
+      headers: {
+        'Authorization' => 'Token 5473850f52c15d4145e168d1e1124dddb42c968f'
```
Comment (orsted51) [inline app/controllers/concerns/http_connection.rb:12]: Not a big deal, but this should never have make it to GitHub, It is recommended to use environment variables instead.

https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository

**Review by orsted51 — CHANGES_REQUESTED** [review]

File: app/controllers/concerns/http_connection.rb (hunk 1 above)
Comment (owl12358) [reply app/controllers/concerns/http_connection.rb]: ahh my mistake. https://github.com/team-ci/messenger/pull/1/commits/a06fee0b797d341ee3c4cbec86a6c0a1d8579294

With this change, we'll need to store the Stackmaps auth token as well as the seeded superuser credentials in the production env. `{superuser: {email: XXX, password: XXX}, stackmaps: {token: XXX}}` @orsted51 

@pseudo132 can you delete the `teacher-lc` account from Stackmaps and create a new one I can use locally? that's the one with the exposed credential.

Comment (pseudo132) [reply app/controllers/concerns/http_connection.rb]: Yes. Complete!
Screen Shot 2022-05-03 at 2 09 00 PM

Comment (pseudo132) [reply app/controllers/concerns/http_connection.rb]: New credentials have been made.

**Review by orsted51 — APPROVED** [review]

**Pull request merged**